TSPL.docc/ReferenceManual/DataRaceSafety.md

```diff
@@ -0,0 +1,774 @@
+# Data-Race Safety
+
+Eliminate data races at compile time.
+
+The Swift 6 language mode eliminates data races at compile time by
+identifying and diagnosing risk of concurrent access to shared state.
+
+## Data Isolation
+
+Swift understands and verifies the safety of all mutable state in your code.
+When you share values between concurrent code, Swift ensures that the value
+is either safe to share, or that it is only accessed by one task at a
+time. This guarantee is called _data isolation_, and it maintains
+mutually exclusive access to mutable state at compile time.
+For an overview of isolation in Swift, see <doc:Concurrency#Isolation>.
+
+### Isolation Domains
+
+Every declaration in Swift code has a specific isolation domain. There
+are three kinds of isolation domains:
+1. Non-isolated
+2. Isolated to an actor instance
+3. Isolated to a global actor
+
+Tasks and actors provide isolation domains, and mutable state must
+only be accessed from one isolation domain at a time.
+You can send mutable state from one isolation domain to another,
+and Swift ensures that mutable state is not accessed from multiple
+isolation domains at once.
+
+### Non-isolated
+
+A non-isolated declaration can run in any isolation domain.
+Non-isolated code can run directly on an actor, or it can run
+outside of any actor.
+
+You specify that a declaration is non-isolated with the `nonisolated`
+keyword. Swift guarantees that non-isolated declarations are safe
+to call from anywhere by ensuring that they do not access actor-isolated
+state in their implementation. Non-isolated functions safely call other
```

Comment:
It's also OK for non-isolated function to access actor properties which are `let`-constants and of Sendable types. I think it might be worth adding this information, because it's common when conforming an actor to protocols.